src/main/java/com/idrsolutions/microservice/utils/SettingsValidator.java

```diff
@@ -0,0 +1,120 @@
+package com.idrsolutions.microservice.utils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Set;
+
+public class SettingsValidator {
+
+    private final StringBuilder errorMessage = new StringBuilder();
+    private final HashMap<String, String> paramMap = new HashMap<>();
+
+    public SettingsValidator(final String[] settings) {
+        if (settings != null) { //handle string based parameters
+            if (settings.length % 2 == 0) {
+                for (int i = 0; i < settings.length; i += 2) {
+                    paramMap.put(settings[i], settings[i + 1]);
+                }
+            } else {
+                errorMessage.append("Invalid length of String arguments, should be key:value; pairs.\n");
+            }
+        }
+    }
+
+    public String validateString(final String setting, final String[] values, final boolean required) {
+        if (paramMap.containsKey(setting)) {
+            final String value = paramMap.remove(setting);
+            if (!Arrays.asList(values).contains(value)) {
+                errorMessage.append(required ? "Required " : "Optional ").append("setting \"").append(setting)
+                        .append("\" has incorrect value. Valid values are ").append(Arrays.toString(values)).append('\n');
+            }
+            return value;
+        } else {
+            if (required) {
+                errorMessage.append("Required setting \"").append(setting).append("\" missing. Valid values are ")
+                        .append(Arrays.toString(values)).append(".\n");
+            }
+        }
+        return null;
+    }
+
+    public String validateString(final String setting, final String regex, final boolean required) {
+        if (paramMap.containsKey(setting)) {
+            final String value = paramMap.remove(setting);
+            if (!value.matches(regex)) {
+                errorMessage.append(required ? "Required " : "Optional ").append("setting \"").append(setting)
+                        .append("\" has incorrect value. Please check the API documents for more details.\n");
+            }
+            return value;
+        } else {
+            if (required) {
+                errorMessage.append("Required setting \"").append(setting).append("\" missing. Valid value must match the regex \"")
+                        .append(regex).append("\".\n");
+            }
+        }
+        return null;
+    }
+
+    public Boolean validateBoolean(final String setting, final boolean required) {
+        if (paramMap.containsKey(setting)) {
+            final String value = paramMap.remove(setting);
+            if (!(value.equalsIgnoreCase("true") | value.equalsIgnoreCase("false"))) {
```

Comment:
```suggestion
            if (!value.equalsIgnoreCase("true") && !value.equalsIgnoreCase("false")) {
```

Reply:
Can I confirm, is this a problem with the existing line not working or for readability?
Currently looks like both lines do the same, wanted to be certain I haven't missed something

Reply:
It's currently doing a bitwise or which is why I spotted it. I think getting rid of the brackets is slightly more readable too.

Reply:
My understanding was in an if statement, single | is OR without shortcut if left value is true.
Admittedly that should still of been ||

Regardless your line is clearer, I'll make the change.

Reply:
👍 That matches my understanding. Functionally the result is the same but I figured the bitwise expression was accidental.

Reply:
Most recent change includes this improvement